app/actions/service_credential_binding_key_create.rb

```diff
@@ -18,17 +18,15 @@ def initialize(user_audit_info, audit_hash)
       PERMITTED_BINDING_ATTRIBUTES = [:credentials].freeze
 
       def precursor(service_instance, message:)
-        validate!(service_instance)
+        key = ServiceKey.first(service_instance: service_instance, name: message.name)
+        validate!(service_instance,key,message)
```

Comment:
Minor thing: passing `message.name` into the `validate!` function is sufficient.